src/tree/FunctionPathsFactory.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * @copyright 2018 Hostnet B.V.
+ */
+declare(strict_types = 1);
+
+class FunctionPathsFactory
+{
+    /**
+     * @param Node $file
+     * @return array
+     */
+    private function produceTokens(Node $file): array
+    {
+        $file_contents = file_get_contents($file->getLocation());
+        $tokens        = token_get_all($file_contents);
+
+        return [
+            "location" => $file->getLocation(),
+            "tokens" => $tokens,
+        ];
+    }
+
+    /**
+     * @param array $file
+     * @return array string
```

Comment:
I think this is an array of `FileFunction` objects. So `@return FileFunction[]` would be the correct typehint